Pylint Spring Cleaning

Eearslya opened the pull request:
This PR addresses the majority of pylint's complaints since the recent configuration change.

There are still a few warnings that I left in, mainly the `too-many-branches` warnings, as I feel that these are candidates for refactoring and should not be suppressed.

There is also an issue of `rule.py` redefining the name `rule`, though I'm not sure if this should be fixed or ignored.

Eearslya committed:
Rename pylint config to .pylintrc

.pylintrc is the default recognized by pylint. We no longer have to specify a config file this way.

Eearslya committed:
Pylint spring cleaning pass one

shatteredbeam reviewed (dismissed):
LGTM

theunkn0wn1 commented on src/packages/commands/rat_command.py:
```diff
@@ -15,39 +15,35 @@
 
 
-_registered_commands = {}
+_REGISTERED_COMMANDS = {}
```
registered commands is by no means a constant, as multiple things modify its contents

theunkn0wn1 commented on src/packages/commands/rat_command.py:
```diff
@@ -108,25 +104,26 @@ def _register(func, names: list or str) -> bool:
+
+    # again this feels ugly but they are module-level now...
+    global _REGISTERED_COMMANDS  # pylint: disable=global-statement
+    _REGISTERED_COMMANDS = {}
```
Since we are modifying this statement anyways in this context we can actually get rid of the global statement.

we can use `_REGISTERED_COMMANDS.clear()` which will get rid of any reason to use the global statement


```py
_REGISTERED_COMMANDS.clear()
```

theunkn0wn1 reviewed (changes requested):
Looking good for the most part, just a couple small things that should be addressed.

Eearslya committed:
Set _registered_commands back to a normal var for now

This will likely be addressed in an upcoming PR, wherein this variable will be moved into a class.

theunkn0wn1 commented on tests/test_rat_command.py:
```diff
@@ -108,7 +108,7 @@ def test_register_non_callable(self, garbage):
             return bot, channel, sender
 
-        assert alias.lower() in Commands._REGISTERED_COMMANDS.keys()
+        assert alias.lower() in Commands._registered_commands.keys()
```
[suggestion] access to `.keys` is redundant. the contains check for dictionary checks against the keys
```suggestion
        assert alias.lower() in Commands._registered_commands
```

theunkn0wn1 reviewed (commented):
nitpick, with a simple resolution (hopefully)
